lib/matplotlib/tri/tricontour.py

```diff
@@ -76,7 +72,8 @@ def _contour_args(self, args, kwargs):
         self.zmax = float(z_check.max())
         self.zmin = float(z_check.min())
         if self.logscale and self.zmin <= 0:
-            raise ValueError('Cannot %s log of negative values.' % fn)
+            raise ValueError('Cannot %s log of negative values.' %
+                             'contourf' if self.filled else 'contour')
```

Comment:
The `%` has higher precedence; you will get just `'contour'` if `self.filled` is false.

Reply:
Good catch! Didn't know that.